embedded-devices/src/devices/texas_instruments/tmp102/mod.rs

```diff
@@ -0,0 +1,136 @@
+//!     .read_temperature()
+//!     .get::<degree_celsius>()
+//!     .to_f32();
+//! println!("Current temperature: {:?}°C", temp);
+//!
+//! // Perform a one-shot measurement now and return to sleep afterwards.
+//! let temp = tmp102.oneshot(&mut Delay).await?.get::<degree_celsius>().to_f32();
+//! println!("Oneshot temperature: {:?}°C", temp);
+//! # Ok(())
+//! # }
+//! ```
+
+use embedded_devices_derive::{device, device_impl};
+use embedded_registers::RegisterInterface;
+use uom::num_rational::Rational32;
+use uom::si::rational32::ThermodynamicTemperature;
+use uom::si::thermodynamic_temperature::degree_celsius;
+
+pub mod address;
+pub mod registers;
+
+type TMP102I2cCodec = embedded_registers::i2c::codecs::OneByteRegAddrCodec;
+
+/// The TMP102 is a general purpose digital temperature sensor. It provides a 13-bit
+/// temperature result with a resolution of 0.0625 °C and an accuracy of up to ±3 °C across the
+/// temperature range of –40 °C to 125 °C with no calibration.
+///
+/// For a full description and usage examples, refer to the [module documentation](self).
+#[device]
+pub struct TMP102<I: RegisterInterface> {
+    /// The interface to communicate with the device
+    interface: I,
+}
+
+crate::simple_device::i2c!(TMP102, self::address::Address, SevenBitAddress, TMP102I2cCodec);
+
+#[device_impl]
+impl<I: RegisterInterface> TMP102<I> {
+    /// Performs a one-shot measurement. This will set `shutdown` in [`self::registers::Configuration´].
+    /// which will cause the device to perform a single conversion a return to sleep mode afterwards.
+    ///
+    /// This function will initialize the measurement, wait until the data is acquired and return
+    /// the temperature.
+    pub async fn oneshot<D: hal::delay::DelayNs>(
+        &mut self,
+        delay: &mut D,
+    ) -> Result<ThermodynamicTemperature, I::Error> {
+        use self::registers::{Configuration, Temperature};
+
+        // Read current configuration to determine conversion ratio and delay
+        let mut reg_conf = self.read_register::<Configuration>().await?;
+        reg_conf.write_oneshot(true);
+
+        // Initiate measurement
+        self.write_register(&reg_conf).await?;
+
+        // Wait for the duration of the conversion
+        let active_conversion_time = reg_conf.read_conversion_cycle_time().conversion_time_ms() + 10;
+        delay.delay_ms(active_conversion_time).await;
+
+        let raw_temp = self.read_register::<Temperature>().await?.read_raw_temperature() as i32;
+        if reg_conf.read_extended() {
+            Ok(ThermodynamicTemperature::new::<degree_celsius>(Rational32::new_raw(
+                raw_temp, 128,
+            )))
+        } else {
+            Ok(ThermodynamicTemperature::new::<degree_celsius>(Rational32::new_raw(
+                raw_temp, 256,
+            )))
+        }
+    }
+
+    /// Read the last temperature measured
+    pub async fn read_temp(&mut self) -> Result<ThermodynamicTemperature, I::Error> {
```

Comment:
Can we call this `read_temperature` to keep it in the same style as used in the other existing drivers?